app/models/timecard_entry.rb

```diff
@@ -1,12 +1,12 @@
 class TimecardEntry < ApplicationRecord
   belongs_to :member
-  belongs_to :eventdate
+  belongs_to :eventdate, required: true
```

Comment:
For some reason this validation isn't triggering. I also tried adding `:timecard_entries` to `validates_associated` in the `eventdate` model. I think we need to add a `before_destroy` to the `eventdate` model.